documents/write_acls.go

```diff
@@ -84,3 +89,121 @@ func newChangedField(p proofs.Property, leaf *proofs.LeafNode, old bool) changed
 	cf.new = v
 	return cf
 }
+
+// transitionRulesFor returns a copy all the transition rules for the account.
+func (cd *CoreDocument) transitionRulesFor(account identity.DID) (rules []coredocumentpb.TransitionRule) {
+	for _, rule := range cd.Document.TransitionRules {
+		for _, rk := range rule.Roles {
+			role, err := getRole(rk, cd.Document.Roles)
+			if err != nil {
+				continue
+			}
+
+			if _, ok := isAccountInRole(role, account); !ok {
+				continue
+			}
+
+			rules = append(rules, coredocumentpb.TransitionRule{
+				RuleKey:   copyBytes(rule.RuleKey),
+				Roles:     copyByteSlice(rule.Roles),
+				MatchType: rule.MatchType,
+				Field:     copyBytes(rule.Field),
+				Action:    rule.Action,
+			})
+		}
+	}
+
+	return rules
+}
+
+func copyBytes(data []byte) []byte {
```

Comment:
maybe these could be moved to utils?

Reply:
thought so but not used elsewhere. lets keep them here until the time we need to move them ther

Reply:
We should aim to not have utils package. it just against Go standards

Reply:
Please move it there or use something already there. I don't think we are ready to move out utils to its own repo yet,
as per https://twitter.com/davecheney/status/697551191752855552?lang=en

Reply:
Unless there is a place to move thse functions there, I would like to keep them here. We may not remove utils package but we wont put there unneeded stuff that is used by the single package. Lets keep the standards community defined rather than defining our own 👍

Reply:
Like the tweet says, we dont want to create a new utils project :)

Reply:
could you refer here to the community convention you follow? I would like to read :)

Reply:
From my experience, there are problems with both approaches,
With utils
- You can announce a reusable function so that it can be reused by everyone else
- utils makes things well tested for all implementors 
Otherwise you idea `No one else needs` becomes a self full-filling prophecy, I like to keep track of reusable functions in utils and test them properly. Also building software projects properly is multi dimensional optimisation problem, what makes sense for a library might not make sense for a daemon like ours.